crates/rerun_c/src/component_type_registry.rs

```diff
@@ -0,0 +1,36 @@
+use once_cell::sync::Lazy;
+use parking_lot::RwLock;
+use re_sdk::ComponentName;
+
+use crate::CComponentTypeHandle;
+
+pub struct ComponentType {
+    pub name: ComponentName,
+    pub datatype: arrow2::datatypes::DataType,
+}
+
+#[derive(Default)]
+pub struct ComponentTypeRegistry {
+    next_id: CComponentTypeHandle,
+    types: Vec<ComponentType>,
+}
+
+impl ComponentTypeRegistry {
+    pub fn register(
+        &mut self,
+        name: ComponentName,
```

Comment:
Is it worth tracking and warning on re-registering with the same name?

Reply:
that though blipped by me but I didn't act on it. Let's add a debug check that does that!

Reply:
I had to type exactly `#[cfg` for copilot to fill out the correct code xD
Wondering if it reads the comments